lib/src/intl_suggestors/utils.dart

```diff
@@ -0,0 +1,346 @@
+import 'package:analyzer/dart/ast/ast.dart';
+import 'package:analyzer/dart/ast/token.dart';
+import 'package:analyzer/dart/element/element.dart';
+import 'package:analyzer/dart/element/type.dart';
+import 'package:file/file.dart';
+import 'package:over_react_codemod/src/util/component_usage.dart';
+import 'package:collection/collection.dart' show IterableExtension;
+
+import 'constants.dart';
+
+// -- Node Validation --
+bool isValidStringInterpolationNode(AstNode node) {
+  if (!(node is StringInterpolation)) return false;
+  //We do not need to localize single values.  This should be handled by the
+  // variable being passed in.  IE: ..foo = '$bar' becomes [', '$bar', ']
+  if (node.elements.length == 3 &&
+      node.elements.first.toString() == node.elements.last.toString())
+    return false;
+  return true;
+}
+
+bool isValidStringLiteralNode(AstNode node) {
+  if (!(node is SimpleStringLiteral)) return false;
+  if (node.value.isEmpty) return false;
+  if (double.tryParse(node.stringValue!) != null) return false;
```

Comment:
#nit there's a non-nullable getter for this
```suggestion
  if (double.tryParse(node.value) != null) return false;
```